tests/test-coauthors-plus.php

```diff
@@ -679,6 +699,1434 @@ public function test_update_author_term_when_author_term_not_exist() {
 		$coauthors_plus->coauthor_taxonomy = $taxonomy_backup;
 	}
 
+	/**
+	 * This is a basic test to ensure that any authors being assigned to a post
+	 * using the CoAuthors_Plus::add_coauthors() method are appropriately
+	 * associated to the post. Some of the things the add_coauthors()
+	 * method should do are:
+	 *
+	 * 1. Ensure that the post_author is set to the first author in the list
+	 * 2. This is done internally by calling CoAuthors_Plus::get_coauthor_by(),
+	 * which should return a WP_User in this instance (since the author is not linked to a coauthor account)
+	 * 3. Since this coauthor is not linked, create the author's coauthor term, and associate it to the post.
+	 *
+	 * @return void
+	 */
+	public function test_assign_post_author_from_author_who_has_not_been_linked() {
+		$post_id = $this->factory()->post->create(
+			array(
+				'post_author' => $this->author2->ID,
+				'post_status' => 'publish',
+				'post_type'   => 'post',
+			)
+		);
+
+		$query1 = new WP_Query(
+			array(
+				'p' => $post_id
+			)
+		);
+
+		$this->assertEquals( 1, $query1->found_posts );
+		$this->assertEquals( $this->author2->ID, $query1->posts[0]->post_author );
+
+		$first_added_authors = $this->_cap->add_coauthors( $post_id, array( $this->author3->user_login ) );
+		$this->assertTrue( $first_added_authors );
+
+		$query2 = new WP_Query(
+			array(
+				'p' => $post_id
+			)
+		);
+
+		$this->assertEquals( 1, $query2->found_posts );
+		$this->assertEquals( $this->author3->ID, $query2->posts[0]->post_author );
+
+		$author3_term = $this->_cap->get_author_term( $this->author3 );
+
+		$this->assertInstanceOf( WP_Term::class, $author3_term );
```

Comment:
> Do we need to some assertions to check the before state to show that author didn't have an author term before they were added as an author to the post?
Yes, I think that's a great idea! How should we go about doing this?

Reply:
Since leaving the comment, I read up more on this _guard assertion_ ([example](https://stackoverflow.com/questions/1021007/should-it-be-arrange-assert-act-assert)), and what that pre-condition is checking that the default state + our arrangement is in the state we think it should be - at which point, it _could_ be a separate test anyway. I'm undecided on going that far, so we can stick to just repeating the regular assertion (but looking for the inverse state) before the Act.

In the basic principle, we'd call try and get `$author_3_term`, and check it is not and instance of WP_Term. Then the Act of calling `add_coauthors()` would be done (line 733), and then this assertion that it was successfully now a WP_Term would stay.